src/mumble/MainWindow.cpp

```diff
@@ -899,43 +899,172 @@ void MainWindow::openUrl(const QUrl &url) {
 	g.sh->start(QThread::TimeCriticalPriority);
 }
 
+class ChannelUrlPathHandler {
+private:
+	static const QString SEP;
+
+	static QString getEncodedChannelname(Channel *c) {
+		return QString::fromUtf8(QUrl::toPercentEncoding(c->qsName));
+	}
+
+	static Channel *searchChannelV1(Channel *parent, const QString &name) {
+		foreach(Channel *c, parent->qlChannels) {
+			// Compare in lower case for backwards compatibility (can lead to
+			// mistakes with channels where only the case is different)
+			if (c->qsName.toLower() == name.toLower()) {
+				return c;
+			}
+		}
+		return NULL;
+	}
+
+	static Channel *searchChannelV1(Channel *parent, QStringList &subPath, const QString &namePrefix, bool &retExactMatch) {
+		// We are out of stuff/path to look for
+		if (subPath.isEmpty()) {
+			// If we have an unidentified channel, we failed to find an exact match
+			retExactMatch = namePrefix.isNull();
+			return parent;
+		}
+
+		// Next channel name to look for
+		QString name = subPath.takeFirst();
+		// If we still have an unidentified channel, construct name with slash
+		if (!namePrefix.isNull()) {
+			name.prepend(SEP);
+			name.prepend(namePrefix);
+		}
+
+		Channel *chan = searchChannelV1(parent, name);
+
+		return searchChannelV1(chan ? chan : parent, subPath, chan ? QString() : name, retExactMatch);
+	}
+
+	// Depth first search (not searching for alternative exact matches that are
+	// possible through ambiguous slashes or case)
+	static Channel *searchChannelV1(Channel *root, const QString &path, bool &retExactMatch) {
```

Comment:
I think these overloads for searchChannelV1 are confusing.
I wouldn't mind them as much if they were named according to the job they do, but the current naming scheme takes a while to understand if you're trying to understand the code.

As a start, how about inlining `static Channel *searchChannelV1(Channel *parent, const QString &name)` into its only call site in `static Channel *searchChannelV1(Channel *parent, QStringList &subPath, const QString &namePrefix, bool &retExactMatch)`?

Then we're down to two searchChannelV1 functions.

Then maybe one of them could be renamed?